### ISSUE TITLE
dartsass source maps sources not being recognized

### ISSUE THREAD

**Issue by faraixyz** [gohugoio/hugo#8961]
I've noticed that the sourcemap sources aren't being picked up when I use dartsass. I've got a [sample repo](https://github.com/faraixyz/broken-dartsass-sourcemap) [site](https://faraixyz.github.io/broken-dartsass-sourcemap/) where I have rendered the sass in dartsass and libsass.

While the sourcemaps work fine in libsass, they don't in dartsass.

Here's what it looks like in libsass

```json
{
	"version": 3,
	"file": "lib.css",
	"sourceRoot": "C:/dev/broken-dartsass-sourcemap",
	"sources": [
		"assets/css/main.scss",
		"assets/css/_base.scss",
		"assets/css/_site-nav.scss",
		"assets/css/_site-header.scss",
		"assets/css/_utilities.scss"
	],
	"sourcesContent": [
		"@import '_base';\r\n@import '_site-nav';\r\n@import '_site-header';\r\n@import '_utilities';\r\n",
		"* ::before, ::after {\r\n    box-sizing: border-box;\r\n}\r\n\r\n:root {\r\n    color-scheme: light dark;\r\n\r\n    // Adapted from https://developer.apple.com/design/human-interface-guidelines/ios/visual-design/color/#dynamic-system-colors\r\n    --dark-gray-6: rgb(15, 22, 43);\r\n    --light-gray-6: rgb(240, 240, 250);\r\n    --light-sys-blue: rgb(0, 64, 221);\r\n    --dark-sys-blue: rgb(64, 156, 255);\r\n    --light-sys-pink: rgb(211, 15, 69);\r\n    --dark-sys-pink: rgb(255, 100, 130);\r\n    --light-sys-indigo: rgb(54, 52, 163);\r\n    --dark-sys-indigo: rgb(125, 122, 255);\r\n    --bg-col: var(--light-gray-6);\r\n    --fg-col: var(--dark-gray-6);\r\n    --unvisited-link-col: var(--light-sys-blue);\r\n    --active-link-col: var(--light-sys-pink);\r\n    --visited-link-col: var(--light-sys-indigo);\r\n\r\n    @media (prefers-color-scheme: dark) {\r\n        --bg-col: var(--dark-gray-6);\r\n        --fg-col: var(--light-gray-6);\r\n        --unvisited-link-col: var(--dark-sys-blue);\r\n        --active-link-col: var(--dark-sys-pink);\r\n        --visited-link-col: var(--dark-sys-indigo);\r\n    }\r\n}\r\n\r\na {\r\n    color: var(--unvisited-link-col);\r\n    \r\n    &:visited {\r\n        color: var(--visited-link-col);\r\n    }\r\n\r\n    &:active {\r\n        color: var(--active-link-col);\r\n    }\r\n\r\n\r\n}\r\n\r\nhtml {\r\n    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;\r\n    font-size: 125%;\r\n    line-height: 1.5;\r\n    background-color: var(--bg-col);\r\n    color: var(--fg-col);\r\n\r\n    @media (prefers-reduced-motion: no-preference) {\r\n        scroll-behavior: smooth;\r\n    }\r\n}\r\n\r\nmain:focus {\r\n    outline: none;\r\n}\r\n\r\nbody {\r\n    max-width: 80ch;\r\n    margin: 0 auto;\r\n    padding: 0.75rem;\r\n}\r\n\r\nimg, svg, video, canvas, audio, iframe, embed, object {\r\n    display: block;\r\n    vertical-align: middle;\r\n    max-width: 100%;\r\n}\r\n\r\naudio:not([controls]) {\r\n    display:none;\r\n}\r\n\r\nimg, svg, video, canvas {\r\n    height: auto;\r\n}\r\n\r\n[hidden] {\r\n    display: none;\r\n}\r\n\r\n.highlight pre {\r\n    padding: 1rem;\r\n    overflow-x: auto;\r\n}\r\n",
		".site-nav__list {\r\n    list-style: none;\r\n    padding-left: 0;\r\n    text-align: center;\r\n    \r\n    li {\r\n        display: inline;\r\n        padding: 0.25rem;\r\n    }\r\n}\r\n",
		".site-header {\r\n    display: flex;\r\n    align-items: center;\r\n    flex-flow: column wrap;\r\n\r\n    * {\r\n        margin: 0;\r\n    }\r\n\r\n    & > * + * {\r\n        margin-top: 0.25rem;\r\n    }\r\n}\r\n\r\n.site-header a[href=\"#main\"] {\r\n    position: absolute;\r\n    top: 0;\r\n    background-color: black;\r\n    color: white;\r\n    padding: 0.25rem;\r\n    outline: none;\r\n    transform: scale(0);\r\n\r\n    &:focus {\r\n        transform: none;\r\n    }\r\n\r\n    @media(prefers-color-scheme: dark) {\r\n        color: black;\r\n        background-color: white;\r\n    }\r\n}\r\n\r\n.site-header strong {\r\n    text-align: center;\r\n}\r\n",
		".over-scroll {\r\n    overflow-x: auto;\r\n    table {\r\n        white-space: nowrap;\r\n    }\r\n}\r\n\r\n.wrap-code-blocks .highlight > pre, .wrap-code-blocks.highlight > pre {\r\n    white-space: pre-wrap !important;\r\n}\r\n\r\nimg.invert-hack {\r\n    @media(prefers-color-scheme: dark) {\r\n        filter: invert(100%);\r\n    }\r\n}\r\n"
	],
	"names": [],
	"mappings": "ACAA,AAAA,CAAC,GAAG,MAAM,IAAI,KAAK,CAAC;EAChB,UAAU,EAAE,UAAU,GACzB;;CAEA,AAAD,IAAK,CAAC;EACF,YAAY,EAAE,UAAU;EAGxB,aAAa,CAAA,gBAAC;EACd,cAAc,CAAA,mBAAC;EACf,gBAAgB,CAAA,gBAAC;EACjB,eAAe,CAAA,kBAAC;EAChB,gBAAgB,CAAA,iBAAC;EACjB,eAAe,CAAA,mBAAC;EAChB,kBAAkB,CAAA,iBAAC;EACnB,iBAAiB,CAAA,mBAAC;EAClB,QAAQ,CAAA,oBAAC;EACT,QAAQ,CAAA,mBAAC;EACT,oBAAoB,CAAA,sBAAC;EACrB,iBAAiB,CAAA,sBAAC;EAClB,kBAAkB,CAAA,wBAAC,GAStB;EAPG,MAAM,6BAlBV;KAAC,AAAD,IAAK,CAAC;MAmBE,QAAQ,CAAA,mBAAC;MACT,QAAQ,CAAA,oBAAC;MACT,oBAAoB,CAAA,qBAAC;MACrB,iBAAiB,CAAA,qBAAC;MAClB,kBAAkB,CAAA,uBAAC,GAE1B,EAAA;AAED,AAAA,CAAC,CAAC;EACE,KAAK,EAAE,yBAAyB,GAWnC;EAZD,AAGI,CAHH,CAGK,OAAO,CAAC;IACN,KAAK,EAAE,uBAAuB,GACjC;EALL,AAOI,CAPH,CAOK,MAAM,CAAC;IACL,KAAK,EAAE,sBAAsB,GAChC;;AAKL,AAAA,IAAI,CAAC;EACD,WAAW,EAAE,2HAA2H;EACxI,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,GAAG;EAChB,gBAAgB,EAAE,aAAa;EAC/B,KAAK,EAAE,aAAa,GAKvB;EAHG,MAAM,wCAPV;IAAA,AAAA,IAAI,CAAC;MAQG,eAAe,EAAE,MAAM,GAE9B,EAAA;AAED,AAAA,IAAI,CAAC,KAAK,CAAC;EACP,OAAO,EAAE,IAAI,GAChB;;AAED,AAAA,IAAI,CAAC;EACD,SAAS,EAAE,IAAI;EACf,MAAM,EAAE,MAAM;EACd,OAAO,EAAE,OAAO,GACnB;;AAED,AAAA,GAAG,EAAE,GAAG,EAAE,KAAK,EAAE,MAAM,EAAE,KAAK,EAAE,MAAM,EAAE,KAAK,EAAE,MAAM,CAAC;EAClD,OAAO,EAAE,KAAK;EACd,cAAc,EAAE,MAAM;EACtB,SAAS,EAAE,IAAI,GAClB;;AAED,AAAA,KAAK,CAAA,GAAK,EAAA,AAAA,QAAC,AAAA,GAAW;EAClB,OAAO,EAAC,IAAI,GACf;;AAED,AAAA,GAAG,EAAE,GAAG,EAAE,KAAK,EAAE,MAAM,CAAC;EACpB,MAAM,EAAE,IAAI,GACf;;CAED,AAAA,AAAA,MAAC,AAAA,EAAQ;EACL,OAAO,EAAE,IAAI,GAChB;;AAED,AAAA,UAAU,CAAC,GAAG,CAAC;EACX,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,IAAI,GACnB;;ACxFD,AAAA,eAAe,CAAC;EACZ,UAAU,EAAE,IAAI;EAChB,YAAY,EAAE,CAAC;EACf,UAAU,EAAE,MAAM,GAMrB;EATD,AAKI,eALW,CAKX,EAAE,CAAC;IACC,OAAO,EAAE,MAAM;IACf,OAAO,EAAE,OAAO,GACnB;;ACRL,AAAA,YAAY,CAAC;EACT,OAAO,EAAE,IAAI;EACb,WAAW,EAAE,MAAM;EACnB,SAAS,EAAE,WAAW,GASzB;EAZD,AAKI,YALQ,CAKR,CAAC,CAAC;IACE,MAAM,EAAE,CAAC,GACZ;EAPL,AASI,YATQ,GASJ,CAAC,GAAG,CAAC,CAAC;IACN,UAAU,EAAE,OAAO,GACtB;;AAGL,AAAA,YAAY,CAAC,CAAC,CAAA,AAAA,IAAC,CAAK,OAAO,AAAZ,EAAc;EACzB,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,CAAC;EACN,gBAAgB,EAAE,KAAK;EACvB,KAAK,EAAE,KAAK;EACZ,OAAO,EAAE,OAAO;EAChB,OAAO,EAAE,IAAI;EACb,SAAS,EAAE,QAAQ,GAUtB;EAjBD,AASI,YATQ,CAAC,CAAC,CAAA,AAAA,IAAC,CAAK,OAAO,AAAZ,EAST,KAAK,CAAC;IACJ,SAAS,EAAE,IAAI,GAClB;EAED,MAAM,6BAbV;IAAA,AAAA,YAAY,CAAC,CAAC,CAAA,AAAA,IAAC,CAAK,OAAO,AAAZ,EAAc;MAcrB,KAAK,EAAE,KAAK;MACZ,gBAAgB,EAAE,KAAK,GAE9B,EAAA;AAED,AAAA,YAAY,CAAC,MAAM,CAAC;EAChB,UAAU,EAAE,MAAM,GACrB;;ACnCD,AAAA,YAAY,CAAC;EACT,UAAU,EAAE,IAAI,GAInB;EALD,AAEI,YAFQ,CAER,KAAK,CAAC;IACF,WAAW,EAAE,MAAM,GACtB;;AAGL,AAAA,iBAAiB,CAAC,UAAU,GAAG,GAAG,EAAE,iBAAiB,AAAA,UAAU,GAAG,GAAG,CAAC;EAClE,WAAW,EAAE,mBAAmB,GACnC;;AAGG,MAAM,6BADV;EAAA,AAAA,GAAG,AAAA,YAAY,CAAC;IAER,MAAM,EAAE,YAAY,GAE3B,EAAA"
}
```
and here it is in dartsass

```json
{
    "version": 3,
    "sourceRoot": "",
    "sources": [
        "file://c/dev/broken-dartsass-sourcemap/assets/css/_base.scss",
        "file://c/dev/broken-dartsass-sourcemap/assets/css/_site-nav.scss",
        "file://c/dev/broken-dartsass-sourcemap/assets/css/_site-header.scss",
        "file://c/dev/broken-dartsass-sourcemap/assets/css/_utilities.scss"
    ],
    "names": [],
    "mappings": "AAAA;EACI;;;AAGJ;EACI;EAGA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EAlBJ;IAmBQ;IACA;IACA;IACA;IACA;;;;AAIR;EACI;;AAEA;EACI;;AAGJ;EACI;;;AAMR;EACI;EACA;EACA;EACA;EACA;;AAEA;EAPJ;IAQQ;;;;AAIR;EACI;;;AAGJ;EACI;EACA;EACA;;;AAGJ;EACI;EACA;EACA;;;AAGJ;EACI;;;AAGJ;EACI;;;AAGJ;EACI;;;AAGJ;EACI;EACA;;;ACvFJ;EACI;EACA;EACA;;AAEA;EACI;EACA;;;ACPR;EACI;EACA;EACA;;AAEA;EACI;;AAGJ;EACI;;;AAIR;EACI;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;;AAGJ;EAbJ;IAcQ;IACA;;;;AAIR;EACI;;;AClCJ;EACI;;AACA;EACI;;;AAIR;EACI;;;AAIA;EADJ;IAEQ"
}
```

### What version of Hugo are you using (`hugo version`)?

<pre>
$ hugo version
hugo v0.87.0-B0C541E4+extended windows/amd64 BuildDate=2021-08-03T10:57:28Z VendorInfo=gohugoio
</pre>

### Does this issue reproduce with the latest release?
Yes

**Comment by faraixyz** [comment]
FWIW, if I build the stylesheets manually with dartsass using `sass --source-map-urls=absolute --embed-sources main.scss main.css`, it works just like in libsass.

**Comment by bep** [comment]
I'm pretty sure I tested this OK at some point ...

**Comment by faraixyz** [comment]
I just noticed that this was fixed in v108 through #10488. Thank you!
